pkg/appfile/parser.go

```diff
@@ -406,3 +435,93 @@ func PrepareProcessContext(k8sClient client.Client, wl *Workload, applicationNam
 	}
 	return pCtx, nil
 }
+
+// GetOutputSecretNames set all secret names, which are generated by cloud resource, to context
+func GetOutputSecretNames(workloads *Workload) (string, error) {
+	secretName, err := getComponentSetting(process.OutputSecretName, workloads.Params)
+	if err != nil {
+		return "", err
+	}
+
+	return fmt.Sprint(secretName), nil
+}
+
+func parseWorkloadK8sSecretSchema(ctx context.Context, c client.Client, namespace string, wl *Workload) ([]process.RequiredSecrets, error) {
+	var requiredSecret []process.RequiredSecrets
+	api, err := utils.GenerateOpenAPISchemaFromDefinition(wl.Name, wl.Template)
+	if err != nil {
+		if !errors.Is(err, errors.Errorf(utils.ErrNoSectionParameterInCue, wl.Name)) {
+			return nil, nil
+		}
+		return nil, err
+	}
+
+	schema, err := utils.ConvertOpenAPISchema2SwaggerObject(api)
+	if err != nil {
+		return nil, err
+	}
+	tag := utils.K8sSecretSchemaTag
+	for k, v := range schema.Properties {
+		description := v.Value.Description
+		if strings.Contains(description, tag) {
+			contextName := strings.Split(description, tag)[1]
+			contextName = strings.TrimSpace(contextName)
+			secretNameInterface, err := getComponentSetting(k, wl.Params)
+			if err != nil {
+				return nil, err
+			}
+			secretName := fmt.Sprint(secretNameInterface)
+			secretData, err := extractSecret(ctx, c, namespace, fmt.Sprint(secretName))
+			if err != nil {
+				return nil, err
+			}
+
+			requiredSecret = append(requiredSecret, process.RequiredSecrets{
+				Name:        secretName,
+				ContextName: contextName,
+				Namespace:   namespace,
+				Data:        secretData,
+			})
+		}
+	}
+	return requiredSecret, nil
+}
+
+func extractSecret(ctx context.Context, c client.Client, namespace, name string) (map[string]interface{}, error) {
+	secretData := make(map[string]interface{})
+	var secret v1.Secret
+	if err := c.Get(ctx, client.ObjectKey{Name: name, Namespace: namespace}, &secret); err != nil {
+		return nil, fmt.Errorf("failed to get secret %s from namespace %s which is required by the component: %w",
+			name, namespace, err)
+	}
+	for k, v := range secret.Data {
+		secretData[k] = string(v)
+	}
+	if len(secretData) == 0 {
+		return nil, fmt.Errorf("data in secret %s from namespace %s isn't available", name, namespace)
+	}
+	return secretData, nil
+}
+
+func getComponentSetting(settingParamName string, params map[string]interface{}) (interface{}, error) {
```

Comment:
the return value is definitly a string, let assert it in the function.

```suggestion
func getComponentSetting(settingParamName string, params map[string]interface{}) (string, error) {
```

Reply:
The settings type is `runtime.RawExtension`, and workload Params type is `map[string]interface{}`. So I try to align the  type.